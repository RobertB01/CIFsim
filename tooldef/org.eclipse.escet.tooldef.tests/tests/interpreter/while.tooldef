//////////////////////////////////////////////////////////////////////////////
// Copyright (c) 2010, 2021 Contributors to the Eclipse Foundation
//
// See the NOTICE file(s) distributed with this work for additional
// information regarding copyright ownership.
//
// This program and the accompanying materials are made available
// under the terms of the MIT License which is available at
// https://opensource.org/licenses/MIT
//
// SPDX-License-Identifier: MIT
//////////////////////////////////////////////////////////////////////////////

int z = 10;
while z > 0:
    outln("%s", z);
    z = z - 1;
end
outln();

z = 10;
while z > 0:
    if z == 5:: break;
    outln("%s", z);
    z = z - 1;
end
outln();

z = 10;
while z > 0:
    if z < 5:: break;
    z = z - 1;
    if z mod 2 == 1:: continue;
    outln("%s", z);
end
outln();

tool t():
    int x = 10;
    while true:
        if x < 5:: return;
        out("%s ", x);
        x = x - 1;
        outln("%s", x);
    end
end
t();
