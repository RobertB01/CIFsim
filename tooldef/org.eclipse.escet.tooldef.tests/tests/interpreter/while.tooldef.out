10
9
8
7
6
5
4
3
2
1

10
9
8
7
6

8
6
4

10 9
9 8
8 7
7 6
6 5
5 4
